### PR TITLE
XM Cyber Integration

### PR DESCRIPTION
**Pull request by matan-xmcyber**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
XM Cyber fetches incidents which are events about risk to the organization and also enriches incidents with data about critical assets at risk.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ X ] No

## Must have
- [ V ] Tests
- [ V ] Documentation

### CODE REVIEW

**Comment by mchasepan** [comment]
@matan-xmcyber awesome, nice working with you. Appreciate the work!

**Review by orlichter1 — CHANGES_REQUESTED** [review]
Please run `demisto-sdk format` on all the pack. Should solve most of my comments.

File: Packs/XMCyber/.secrets-ignore [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            "172.0.100.5",
+            "192.168.111.185"
+        ],
+        "urls": [
+            "test.com",
+            "xsoar.pan.dev",
+            "attack.mitre.org",
+            "deployer@domaine.model.cyberxm.com",
+            "xmcyber.example.com",
+            "matan@xmcyber.com",
```
[the remark this answers is not in the document]
Comment (matan-xmcyber) [reply Packs/XMCyber/.secrets-ignore:16]: yes comfortable with it

File: Packs/XMCyber/Integrations/XMCyberIntegration/upload_version.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (matan-xmcyber) [reply Packs/XMCyber/Integrations/XMCyberIntegration/upload_version.sh:1]: I will remove

File: Packs/XMCyber/Integrations/XMCyberIntegration/XMCyberIntegration.yml [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+  type: 0
+
+script:
+  commands:
+  - name: xmcyber-affected-critical-assets-list
+    description: List critical assets at risk from an entity and the complexity of the attack
+    arguments:
+      - name: timeId
+        default: false
+        description: timeId for example timeAgo_days_7
```
[the remark this answers is not in the document]
Comment (matan-xmcyber) [reply Packs/XMCyber/Integrations/XMCyberIntegration/XMCyberIntegration.yml:57]: thanks I elaborated in the description

File: Packs/XMCyber/Playbooks/XM_Cyber_Endpoint_Enrichment_README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+| Endpoint.Domain | Domain of the endpoint | string |
+| Endpoint.MACAddress | MACAddress of the endpoint | string |
+| Endpoint.OS | OS of the endpoint | string |
+| Endpoint.OSVersion | OSVersion of the endpoint | string |
+| Endpoint.BIOSVersion | BIOSVersion of the endpoint | string |
+| Endpoint.HostName | The host of the endpoint | string |
+
+## Playbook Image
+---
+![XM Cyber Endpoint Enrichment](Insert the link to your image here)
```
[the remark this answers is not in the document]
Comment (matan-xmcyber) [reply Packs/XMCyber/Playbooks/XM_Cyber_Endpoint_Enrichment_README.md:46]: I think we'll use the same image as the integration throughout. Does that make sense?

File: Packs/XMCyber/Integrations/XMCyberIntegration/XMCyberIntegration.yml [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+        required: false
+        secret: false
+      - name: entityId
+        default: false
+        description: Entity ID
+        isArray: false
+        required: true
+        secret: false
+    outputs:
+      - contextPath: entityId
```
[the remark this answers is not in the document]
Comment (matan-xmcyber) [reply Packs/XMCyber/Integrations/XMCyberIntegration/XMCyberIntegration.yml:97]: XMCyber.entityId is what it is supposed to be. Thanks

File: Packs/XMCyber/Playbooks/XM_Cyber_Risk_Score_To_Jira.yml [hunk 6]
```diff
@@ -0,0 +1,101 @@
+id: XM_Cyber_Risk_Score_To_Jira
```
[the remark this answers is not in the document]
Comment (matan-xmcyber) [reply Packs/XMCyber/Playbooks/XM_Cyber_Risk_Score_To_Jira.yml:1]: Yes. We agree it seems too simple. Maybe you can help us think of how to add more sophistication to it.

File: Packs/XMCyber/Integrations/XMCyberIntegration/XMCyberIntegration.yml [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+      - name: timeId
+        default: false
+        description: timeId for example timeAgo_days_7
+        isArray: false
+        required: false
+        secret: false
+      - name: entityId
+        default: false
+        description: Entity ID
+        isArray: false
```
[the remark this answers is not in the document]
Comment (matan-xmcyber) [reply Packs/XMCyber/Integrations/XMCyberIntegration/XMCyberIntegration.yml:93]: ok

**Comment by orlichter1** [comment]
So just writing down everything we said:
**Classifier**:
- [ ] remove unused incident type
- [ ] use "Direct unclassified events to" option

**Mapper**
- [ ] remove unused incident field
- [ ] rename the incident type with the link to XM
- [ ]  map fields to relevant XM incident types (can use the "copy from option")

**Jira Playbook**
- [ ] remove hardcoded project
- [ ] remove undersocres 

**Enrichment Playbook**
- [ ] add "isIntegrationConfigured" at the beginning to check that XMCyber integration is configured
- [ ] remove mapper script and use inputs
- [ ] merge with endpoint enrichment playbook

**Playbooks in general**
- [ ] use playbook inputs
- [ ] if you like to rename the playbooks as ".... - XM Cyber"

**Commands**
- [ ] add an `Entity` level to outputs (`XMCyber.Entity....`)
- [ ] remove `xmcyber-entity-get` as it isn't needed anymore

**Comment by matan-xmcyber** [comment]
> So just writing down everything we said:
> **Classifier**:
> 
> * [ x ]  remove unused incident type
> * [ x ]  use "Direct unclassified events to" option
> 
> **Mapper**
> 
> * [ x ]  remove unused incident field
> * [ x ]  rename the incident type with the link to XM
> * [ x ]   map fields to relevant XM incident types (can use the "copy from option")
> 
> **Jira Playbook**
> 
> * [ x ]  remove hardcoded project
> * [ x ]  remove undersocres
> 
> **Enrichment Playbook**
> 
> * [ x ]  add "isIntegrationConfigured" at the beginning to check that XMCyber integration is configured
> * [ x ]  remove mapper script and use inputs
> * [ x ]  merge with endpoint enrichment playbook
> 
> **Playbooks in general**
> 
> * [ x ]   use playbook inputs
> * [ x ]  if you like to rename the playbooks as ".... - XM Cyber"
> 
> **Commands**
> 
> * [ x ]  add an `Entity` level to outputs (`XMCyber.Entity....`)
> * [ x ]  remove `xmcyber-entity-get` as it isn't needed anymore

All is completed

**Comment by orlichter1** [comment]
@matan-xmcyber @MarkRachmanXMCyber Hey guys looks great! Just 3 very small question:
1. We now have 4 different incident types. Is that on purpose? Before we had 3 and we said we don't need one of them.
2. Looks like some unittests are failing for some reason.
3. Did we always have layouts or are they new?

**Comment by matan-xmcyber** [comment]
> @matan-xmcyber @MarkRachmanXMCyber Hey guys looks great! Just 3 very small question:
> 
> 1. We now have 4 different incident types. Is that on purpose? Before we had 3 and we said we don't need one of them.
> 2. Looks like some unittests are failing for some reason.
> 3. Did we always have layouts or are they new?

1. Yes I changed to 4 different incident types, it makes sense for them to be more specific.
2. Unitests should pass now
3. We always had layouts

**Review by orlichter1 — APPROVED** [review]

**Pull request merged**